src/services/project-directory-finder.js

```diff
@@ -0,0 +1,47 @@
+class ProjectDirectoryFinder {
```

Comment:
Same code as the deleted util `ProjectDirectoryUtils` with:
- "full" dependency injection (`path`)
- `pathSchemaDir` option to configure a custom schema directory